

<!DOCTYPE HTML>
<head>
<!--PAGE TITLE-->
<title>Conclusion - Data Analysis Project - Matthew Jackson</title>

<link href='http://fonts.googleapis.com/css?family=Yanone+Kaffeesatz' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=PT+Sans+Caption' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=PT+Sans:400,700' rel='stylesheet' type='text/css'>
<link href='http://fonts.googleapis.com/css?family=Cantarell' rel='stylesheet' type='text/css'>

<link href="//netdna.bootstrapcdn.com/font-awesome/3.1.1/css/font-awesome.css" rel="stylesheet">
<link rel="stylesheet" type="text/css" href="css/main.css">
<link rel="stylesheet" type="text/css" href="css/font-awesome.css">
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.2/jquery.min.js"></script>

</head>



<body>
<script type="text/javascript">
$(document).ready(function() {
	var s = $("#Bsidebar");
	var pos = s.position();
	var s2 = $("#mainrightbar");
	var pos2 = s2.position();					   
	$(window).scroll(function() {
		var windowpos = $(window).scrollTop();
		if (windowpos >= pos.top) {
			s.addClass("stick");
		} else {
			s.removeClass("stick");	
		}
	
		var windowpos = $(window).scrollTop();
		if (windowpos >= pos.top) {
			s2.addClass("stick2");
		} else {	
			s2.removeClass("stick2");
		}
	});

});
</script>
<script type="text/javascript">

function enlargeImg(ima, height, width) {
	myImage = document.getElementById(ima);
    	myImage.style.width = width;
    	myImage.style.height = height;
    myImage.style.cssFloat="none";
    
    	
}

function reduceImg(ima, height, width) {
	myImage = document.getElementById(ima);
    	myImage.style.width = width;
    	myImage.style.height = height;
    	myImage.style.cssFloat="left";
}

</script>


<div id="topbar">
</div>

<div id="Abackground">
	<div id="Agradient">
		<div id="Acontainer">
			<div id="Aleftholder">
			<img src="images/nitrlogo.png" id="nitrologo" />
			</div><!--Aleftholder-->
			<div id="Arightholder">
				<div id="Arightholdertop">
					<div id="Arightholdertopleft">
					A Random Forest-Based Approach to the Discovery of Nitrogen Storage Proteins in <i>M. smegmatis</i>
					</div><!--Arightholdertopleft-->
					<div id="Arightholdertopright">
						<div id="Aauthorlogo">
						<i class="icon-user"></i>
						</div><!--Aauthorlogo-->
						<div id="Aauthorname">
						Matthew Jackson
						</div><!--Aauthorname-->
						<div id="Adataproj">
						DATA ANALYSIS PROJECT
						</div><!--Adataproj-->
					</div><!--Arightholdertopright-->
				</div><!--Arightholdertop-->
				<div id="Arightholderbottom">
					<img src="images/ribbon.png" />
				</div><!--Arightholderbottom-->
			</div><!--Arightholder-->
	
		</div><!--Agradient-->
	</div><!--Acontainer-->
</div><!--Abackground-->

<div id="Bbackground">
	<div id="Bcontainer">
		<div id="Bsidebar">
			<div id="sidebarspacer">
			</div><!--sidebarspacer-->
			
			

			
			<!--REAL MENU-->
			
			<a href="index.html"><div class="inactiveEntry">
				<div class="inactiveIcon">
				<i class="icon-home"></i>
				</div><!--inactiveIcon-->
				<div class="inactiveText">
				Abstract
				</div><!--inactiveText-->
			</div><!--inactiveEntry--></a>
			
			<a href="summary.html"><div class="inactiveEntry">
				<div class="inactiveIcon">
				<i class="icon-globe"></i>
				</div><!--inactiveIcon-->
				<div class="inactiveText">
				Summary
				</div><!--inactiveText-->
			</div><!--inactiveEntry--></a>
			
			<a href="introduction.html"><div class="inactiveEntry">
				<div class="inactiveIcon">
				<i class="icon-info-sign"></i>
				</div><!--inactiveIcon-->
				<div class="inactiveText">
				Introduction
				</div><!--inactiveText-->
			</div><!--inactiveEntry--></a>
			
			
			<a href="results_and_discussion.html"><div class="inactiveEntry">
				<div class="inactiveIcon">
				<i class="icon-bar-chart"></i>
				</div><!--inactiveIcon-->
				<div class="inactiveText" style="padding-top:7px;font-size:16px;line-height:100%;">
				Results and Discussion
				</div><!--inactiveText-->
			</div><!--inactiveEntry--></a>
			
	
			
			
						<!--ACTIVE EXAMPLE-->
			<a href="conclusion.html"><div class="activeEntry" id="entry6purple">
				<div class="activeIcon">
				<i class="icon-lightbulb"></i>
				</div><!--activeIcon-->
				<div class="activeText">
				Conclusion
				</div><!--activeText-->
				<div class ="activeArrow">
				</div><!--activeArrow-->
			</div><!--activeEntry--></a>
			
						<a href="methods.html"><div class="inactiveEntry">
				<div class="inactiveIcon">
				<i class="icon-beaker"></i>
				</div><!--inactiveIcon-->
				<div class="inactiveText">
				Methods
				</div><!--inactiveText-->
			</div><!--inactiveEntry--></a>
			
	
			<a href="references.html"><div class="inactiveEntry">
				<div class="inactiveIcon">
				<i class="icon-book"></i>
				</div><!--inactiveIcon-->
				<div class="inactiveText">
				References
				</div><!--inactiveText-->
			</div><!--inactiveEntry--></a>


			<a href="acknowledgements.html"><div class="inactiveEntry">
				<div class="inactiveIcon">
				<i class="icon-group"></i>
				</div><!--inactiveIcon-->
				<div class="inactiveText" style="font-size:14px;">
				Acknowledgements
				</div><!--inactiveText-->
			</div><!--inactiveEntry--></a>
						
			<a href="supplementary_info.html"><div class="inactiveEntry">
				<div class="inactiveIcon">
				<i class="icon-expand-alt" style="padding-left:7px; width:28px;"></i>
				</div><!--inactiveIcon-->
				<div class="inactiveText" style="padding-top:7px;font-size:16px;line-height:100%;">
				Supplementary Information
				</div><!--inactiveText-->
			</div><!--inactiveEntry--></a>
		
		</div><!--Bsidebar-->
		
		
		
<!--MAIN CONTENT OF PAGE-->		
		<div id="maincontent">
			<div id="maincontentinner">
				<div id="mainleftbar">
				<div class="heading">
				Conclusion
				</div><!--heading-->

<!--PARAGRAPH WITH HEADER-->	


<h1 class="subpurple">Main Conclusion</h1>
<div class="subline"></div>
We have reported a random forest based approach for the prediction of nitrogen storage proteins from sequence, using sequence-derived properties. High accuracies (up to 90%) for the model are not due to protein family bias and show this is a very useful method for predicting nitrogen storage proteins from primary sequence information. 
</p>
<p>Our analysis suggests storage proteins are defined by a general combination of physico-chemical and compositional features rather than a few very important ones. These factors include having fewer charged amino acids (like arginine) and having low length and molecular weight.
</p><p>
The classifier has given additional evidence to the existence of two prime candidates of storage proteins in <i>M.smegmatis</i> (<a href="http://www.uniprot.org/uniprot/A0QTX5" class="weblink" target="_blank">MSMEG_1999</a> and <a href="http://www.uniprot.org/uniprot/A0QQU4" class="weblink" target="_blank">MSMEG_0879</a>) and highlighted a new list of candidates for further investigation. The results lend support to the hypothesis that <i>M.smegmatis</i> may be storing nitrogen in the form of specialised proteins.
</p>

<h1 class="subpurple">How general are the results?</h1>

Although positive set sub-structure was shown by the similarity network and hinted at by the PCA, the classifier remained highly effective at discriminating the storage proteins from a family that it had not been trained on, maintaining accuracy near 90%. 
</p><p>This accuracy was in the context of the model's training and testing protein sets being derived from very diverse species, from microbes to mice, meaning in turn that we can expect the classifier to work well on any species. 

<h1 class="subpurple">Were there problems with the procedures?</h1>
Despite the reported accuracy and lack of bias, there was redundancy within the positive set. For instance, there are 89 entries from the cupin-like family, which are very highly connected in the protein similarity network. It would be good to reduce redundancy before training to see if this would improve the performance of the model.
</p><p>
Before testing the classifier on proteomes, it would be sensible to filter out sequence fragments, as seen in 2 of the 5 top predictions in <i>O.anatinus</i>, as these are unlikely to be storage proteins (storage proteins are generally short <i>complete</i> proteins). There was insufficient time to write and run software to do this in this investigation.

<h1 class="subpurple">Future Work</h1>
<div class="subline"></div>
Firstly, the functional annotations of the potential positive candidates could be investigated in in depth and analysed. Within the group there may prove candidates worthy of experimental investigation, for instance through over expression or gene knockout experiments, to see if this alters the nitrogen-starvation resistance of bacteria.
</p><p>
Secondly, the speed and effectiveness of the random forest suggest a web server could be built to allow scientists to use this software more generally, inputting a positive set of proteins with a function of interest to generate a classifier. This would save others a great deal of time in creating the data-processing tools, and would be in-keeping with the collaborative spirit of bioinformatics and systems biology.




			</div><!--mainleftbar-->
			
			
			</div><!--maincontentinner--> 
		</div><!--maincontent-->
	</div><!--Bcontainer-->
</div><!--Bbackground-->

	<div class="footerspacer">
	</div><!--footerspacer-->

